cli/tests/pcluster/utils/test_pcluster_utils.py

```diff
@@ -667,3 +667,27 @@ def test_is_hit_enabled_cluster(scheduler, expected_is_hit_enabled):
 )
 def test_get_bucket_url(region, expected_url):
     assert_that(get_bucket_url(region)).is_equal_to(expected_url)
+
+
+current_version = "aws-parallelcluster-{0}".format(utils.get_installed_version())
+
+
+@pytest.mark.parametrize(
+    "ami_name, error_expected",
+    [
+        # Compatible ami name
+        ("ami-xxx{0}xxx".format(current_version), False),
+        ("ami-123{0}321".format(current_version), False),
+        ("ami-{0}".format(current_version), False),
+        ("ami-name-one", False),
+        # Incompatible ami name
+        ("ami-aws-parallelcluster-0.0.0", True),
+    ],
+)
+def test_validate_pcluster_version_based_on_ami_name(ami_name, error_expected):
+    if error_expected:
+        with pytest.raises(SystemExit) as sysexit:
+            utils.validate_pcluster_version_based_on_ami_name(ami_name)
+        assert_that(sysexit.value.code).is_not_equal_to(0)
```

Comment:
See if you can match the error message instead of checking the error code here, with something like`pytest.raises(SystemExit, match=expected_message)`